File: 프로그래머스/1/12931. 자릿수 더하기/자릿수 더하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int n) {
    int answer = 0;
    int i=100000000;
    while(i!=0){
        answer+=(n/i);
        n%=i;
        if(i==1) i==0;
        i/=10;
    }
    return answer;
}

File: 프로그래머스/0/120906. 자릿수 더하기/자릿수 더하기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int i=n;
        int temp=1000000;
        boolean cnt=false;
        while(true){
            if(0<=n/temp && n/temp<10) break;
            temp/=10;
        }
        do{
            if(temp==1){ 
                cnt=true;
            }
            answer+=i/temp;
            i%=temp;
            temp/=10;
            if(cnt) break;
        }while(true);
        return answer;
    }
}